notebooks/least_core_basic.ipynb

```diff
@@ -0,0 +1,811 @@
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>0</th>\n",
+       "      <th>1</th>\n",
+       "      <th>2</th>\n",
+       "      <th>3</th>\n",
+       "      <th>4</th>\n",
+       "      <th>5</th>\n",
+       "      <th>6</th>\n",
+       "      <th>7</th>\n",
+       "      <th>8</th>\n",
+       "      <th>9</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>exact_value</th>\n",
+       "      <td>0.04</td>\n",
+       "      <td>0.031579</td>\n",
+       "      <td>0.086842</td>\n",
+       "      <td>0.053158</td>\n",
+       "      <td>0.028947</td>\n",
+       "      <td>0.04</td>\n",
+       "      <td>0.057895</td>\n",
+       "      <td>0.047895</td>\n",
+       "      <td>0.037368</td>\n",
+       "      <td>0.055263</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>exact_value_stderr</th>\n",
+       "      <td>0.00</td>\n",
+       "      <td>0.000000</td>\n",
+       "      <td>0.000000</td>\n",
+       "      <td>0.000000</td>\n",
+       "      <td>0.000000</td>\n",
+       "      <td>0.00</td>\n",
+       "      <td>0.000000</td>\n",
+       "      <td>0.000000</td>\n",
+       "      <td>0.000000</td>\n",
+       "      <td>0.000000</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                       0         1         2         3         4     5  \\\n",
+       "exact_value         0.04  0.031579  0.086842  0.053158  0.028947  0.04   \n",
+       "exact_value_stderr  0.00  0.000000  0.000000  0.000000  0.000000  0.00   \n",
+       "\n",
+       "                           6         7         8         9  \n",
+       "exact_value         0.057895  0.047895  0.037368  0.055263  \n",
+       "exact_value_stderr  0.000000  0.000000  0.000000  0.000000  "
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "exact_values_df = (\n",
+    "    exact_least_core(\n",
+    "        u=utility,\n",
+    "        progress=True,\n",
+    "    )\n",
+    "    .to_dataframe(column=\"exact_value\")\n",
+    "    .T\n",
+    ")\n",
+    "exact_values_df = exact_values_df[sorted(exact_values_df.columns)]\n",
+    "exact_values_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "1091849a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 11.4 s, sys: 914 ms, total: 12.3 s\n",
+      "Wall time: 1min 52s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "budget_array = np.linspace(200, 2 ** (len(small_dataset) - 1), num=10, dtype=int)\n",
+    "\n",
+    "all_estimated_values_df = []\n",
+    "all_errors = {}\n",
+    "\n",
+    "for budget in budget_array:\n",
```

Comment:
can you put a loading bar in the cells that take long? Mostly, those that take >1 minute. Basically, all those that loop over budget array

Reply:
I didn't want to add progress bars there because their output is not saved and they would show up in the documentation as having 0 progress.

I would rather add a note above those cells to warn the readers about those cells taking a long time to finish.

Reply:
On second thought, you're right the progress bars make more sense than a note.